src/sentry/db/deletion.py

```diff
@@ -69,12 +69,38 @@ def execute_generic(self, chunk_size=100):
             else:
                 qs = qs.filter(project_id=self.project_id)
 
+        return self._continuous_generic_query(qs, chunk_size)
+
+    def execute_sharded(self, total_shards, shard_id, chunk_size=100):
+        assert total_shards > 1
+        assert shard_id < total_shards
+        qs = self.model.objects.all().extra(where=[
+            'id %% {total_shards} = {shard_id}'.format(
+                total_shards=total_shards,
+                shard_id=shard_id,
+            )
+        ])
+
+        if self.days:
+            cutoff = timezone.now() - timedelta(days=self.days)
+            qs = qs.filter(
+                **{'{}__lte'.format(self.dtfield): cutoff}
+            )
+        if self.project_id:
+            if 'project' in self.model._meta.get_all_field_names():
+                qs = qs.filter(project=self.project_id)
```

Comment:
Ah. hmm. I'm just surprised this works. I thought one has to do `project_id` or `project__pk` here. I assume this is tested then.